pyk/src/pyk/cli/args.py

```diff
@@ -475,6 +477,9 @@ def kompile_args(self) -> ArgumentParser:
             action='store_true',
             help='Do not wrap the output on the CLI.',
         )
+        args.add_argument(
+            '--ignore-warnings', dest='ignore_warnings', default=None, action='append', help='Ignore provided warnings'
```

Comment:
For compatibility with the `kompile` CLI:

```suggestion
            '--ignore-warnings', '-Wno', dest='ignore_warnings', default=None, action='append', help='Ignore provided warnings'
```

Reply:
Actually, `append` might not be suitable here. Does `kompile` take the option once with a comma-separated list:

```
-Wno foo,bar
```

or multiple times with atomic values:

```
-Wno foo -Wno bar
```

(or perhaps even a combination of those.)

Reply:
If we want to be compatible with the `kompile` CLI, we need to support the way `kompile` consumes this option.

@JuanCoRo, can you check which of the above methods `kompile` supports to take multiple warnings to ignore?

Reply:
Both versions, or a mix of the two, are supported:
```console
$ kompile --version
K version:    v7.1.21
Build date:   Sat Jun 22 00:00:14 BST 2024
$ cat test.k
module TEST
  imports INT

  syntax Int ::= f(Int) [function, total, deprecated]
  rule f(0) => 0
endmodule
$ kompile test.k
[Warning] Compiler: Non exhaustive match detected:
`f(_)_TEST_Int_Int`(#token("1","Int"))
	Source(/Users/brucecollie/code/scratch/test.k)
	Location(4,18,4,54)
	4 |	  syntax Int ::= f(Int) [function, total, deprecated]
	  .	                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[Warning] Compiler: Use of deprecated production found; this syntax may be
removed in the future.
	Source(/Users/brucecollie/code/scratch/test.k)
	Location(5,8,5,12)
	5 |	  rule f(0) => 0
	  .	       ^~~~
[Warning] Compiler: Could not find main syntax module with name TEST-SYNTAX in
definition.  Use --syntax-module to specify one. Using TEST as default.
$ kompile test.k -Wno non-exhaustive-match -Wno missing-syntax-module,deprecated-symbol
$
```

Reply:
This probably calls for a custom action that instead of `append`ing to the list `extend`s it. I'm not sure how well our current CLI convention supports this though.

Anyways, this doesn't have to be addressed in this PR. @JuanCoRo, could you open an issue for tracking this?

Reply:
I don't follow what the problematic behavior here is. Could you elaborate a bit for me to see it @tothtamas28? Thanks!

Reply:
```
-Wno non-exhaustive-match -Wno missing-syntax-module,deprecated-symbol
```

will parse to

```python3
['non-exhaustive-match', 'missing-syntax-module,deprecated-symbol']
```

which is fine in the sense that concatenating them with `,` will give you something you can pass to `kompile`. Would still be nice to parse the atoms separately:

```python
['non-exhaustive-match', 'missing-syntax-module', 'deprecated-symbol']
```

Reply:
Just opened #4479